src/QafooLabs/Bundle/NoFrameworkBundle/EventListener/ViewListener.php

```diff
@@ -57,35 +41,60 @@ public function onKernelView(GetResponseForControllerResultEvent $event)
         }
 
         $controller = $request->attributes->get('_controller');
-        $templateView = $event->getControllerResult();
+        $result = $event->getControllerResult();
 
-        if (!$controller || $templateView instanceof Response) {
+        if (!$controller || $result instanceof Response) {
             return;
         }
 
-        if ( ! ($templateView instanceof TemplateView)) {
-            $templateView = new TemplateView($templateView);
+        $response = ($result instanceof Generator)
+            ? $this->unrollGenerator($result, $request)
+            : $this->convert($result, $request);
+
+        if ($response) {
+            $event->setResponse($response);
         }
+    }
+
+    private function unrollGenerator(Generator $generator, Request $request): ?Response
```

Comment:
I think a nullable return type declaration is not needed here, what do you think?